src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/LoopDetector.kt

```diff
@@ -93,46 +106,54 @@ internal class LoopDetector(
     private val interleavingsLeadToSpinLockSet = InterleavingSequenceTrackableSet()
 
     /**
-     * Helps to determine does current interleaving equal to some saved interleaving leading to spin cycle or not
+     * Helps to determine does current interleaving equal to some saved interleaving leading to spin cycle or not.
      */
     private val loopTrackingCursor = interleavingsLeadToSpinLockSet.cursor
 
-    private var totalExecutionsCount = 0
-
-    private val firstThreadSet: Boolean get() = lastExecutedThread != -1
-
     /**
      * Delegate helper, active in replay (trace collection) mode.
      * It just tracks executions and switches and helps to halt execution or switch in case of spin-lock early.
      */
     private var replayModeLoopDetectorHelper: ReplayModeLoopDetectorHelper? = null
 
-    private val replayModeCurrentCyclePeriod: Int
-        get() = replayModeLoopDetectorHelper?.currentCyclePeriod ?: 0
-
-    val replayModeEnabled: Boolean
-        get() = replayModeLoopDetectorHelper != null
-
     /**
-     * Indicates if we analyze method parameters to calculate the spin-cycle period.
+     * Indicates whether replay mode is currently enabled in the `LoopDetector`.
      */
-    private var mode: Mode = Mode.DEFAULT
+    val replayModeEnabled: Boolean
+        get() = replayModeLoopDetectorHelper != null
 
     /**
      * Indicates that we are in a spin cycle iteration now.
-     * Should be called only in replay mode.
      */
     val replayModeCurrentlyInSpinCycle: Boolean get() =
         replayModeLoopDetectorHelper?.currentlyInSpinCycle ?: false
 
+    /**
+     * In replay mode, represents the period of spin-cycle if spin-cycle was entered.
+     */
+    private val replayModeCurrentCyclePeriod: Int
+        get() = replayModeLoopDetectorHelper?.currentCyclePeriod ?: 0
+
+    /**
+     * Is called before each interleaving processing
+     */
+    fun initialize() {
```

Comment:
Nitpick: wouldn't `clear` or `reset` better describe the purpose of this method? Idem for other places with similar name, that reset state.